SayTheirNames/Source/Controller/Home/Person/PersonController.swift

```diff
@@ -1,30 +1,221 @@
+    }
+    
+    @objc func shareAction(_ sender: Any) {
+        // Present share bottom sheet view
+    }
+    
+    private func registerCells(to tableView: UITableView) {
+        PersonCellType.allCases.forEach { $0.register(to: tableView) }
+    }
+    
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.separatorStyle = .none
+        tableView.allowsSelection = false
+        tableView.insetsContentViewsToSafeArea = false
+        view.addSubview(tableView)
+        
+        NSLayoutConstraint.activate([
+            tableView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 0),
+            tableView.topAnchor.constraint(equalTo: view.topAnchor, constant: 0),
+            tableView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: 0)
+        ])
+        
+        registerCells(to: tableView)
     }
     
-    @IBAction func didPressCloseButton() {
-        self.dismiss(animated: true, completion: nil)
+    private func setupDonationButton() {
+        donationButtonContainerView.setButtonPressed {
+            // Button Action Here
+        }
+        
+        donationButtonContainerView.translatesAutoresizingMaskIntoConstraints = false
+        donationButtonContainerView.backgroundColor = .white
+        view.addSubview(donationButtonContainerView)
+        donationButtonContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+        donationButtonContainerView.leftAnchor.constraint(equalTo: view.leftAnchor).isActive = true
+        donationButtonContainerView.rightAnchor.constraint(equalTo: view.rightAnchor).isActive = true
+        donationButtonContainerView.heightAnchor.constraint(equalToConstant: 90).isActive = true
+        
+        // Now we can set the anchor for the UITableView
+        tableView.bottomAnchor.constraint(equalTo: donationButtonContainerView.topAnchor, constant: 0).isActive = true
     }
+}
 
-    @IBAction func didPressReadButton(_ sender: Any) {
-        let personDetailsController = PersonDetailsController(service: self.service)
-        personDetailsController.isModalInPresentation = true
-        navigationController?.pushViewController(personDetailsController, animated: true)
-      }
+extension PersonController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return cellCollectionTypes.count
+    }
     
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellType = cellCollectionTypes[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellType.identifier, for: indexPath)
+        
+        switch cellType {
+        case .photo:
+            let photoCell = cell as! PersonPhotoTableViewCell
```

Comment:
Probably need to rebase with **development**. Linter should be catching force unwrapped variables.